bokeh/charts/utils.py

```diff
@@ -38,3 +44,109 @@ def polar_to_cartesian(r, start_angles, end_angles):
+            self.doc.add(VBox(children=self.children))
+
+        self.plot = None
+        xdr, ydr = None, None
+        for i, chart in enumerate(self.charts):
+            chart.doc = self.doc
+            if self.server:
+                chart.session = self.session
+
+            # Force the chart to create the underlying plot
+            chart._setup_show()
+            chart._prepare_show()
+            chart._show_teardown()
+
+            if not self.title:
+                self.title = chart.chart.title
+
+        # reset the pot title with the one set for the Figure
+        self.doc._current_plot.title = self.title
+
+    def show(self):
+        """Main show function.
+
+        It shows the Figure in file, server and notebook outputs.
+        """
+        if self.filename:
+            if self.filename is True:
+                filename = "untitled"
+            else:
+                filename = self.filename
+            with open(filename, "w") as f:
+                f.write(file_html(self.doc, INLINE, self.title))
+            print("Wrote %s" % filename)
+            view(filename)
+
+        elif self.filename is False and self.server is False and self.notebook is False:
+            print("You have to provide a filename (filename='foo.html' or"
+                  " .filename('foo.html')) to save your plot.")
+
+        if self.server:
+            self.session.store_document(self.doc)
+            link = self.session.object_link(self.doc.context)
+            view(link)
+
+        if self.notebook:
+            from bokeh.embed import notebook_div
+            for plot in self._plots:
+                publish_display_data({'text/html': notebook_div(plot)})
+
+
+def show(obj, title='test', filename=False, server=False, notebook=False, **kws):
+    """ 'shows' a plot object, by auto-raising the window or tab
+    displaying the current plot (for file/server output modes) or displaying
+    it in an output cell (IPython notebook).
+
+    Args:
+        obj (Widget/Plot object, optional): it accepts a plot object and just shows it.
+
+    """
+    if isinstance(obj, Figure):
+        return obj.show()
+
+    if filename:
+        if filename is True:
+            filename = "untitled"
+        else:
+            filename = filename
+        with open(filename, "w") as f:
+            f.write(file_html(obj, INLINE, title))
+        print("Wrote %s" % filename)
+        view(filename)
+
+    elif filename is False and server is False and notebook is False:
+        print("You have to provide a filename (filename='foo.html' or"
+              " .filename('foo.html')) to save your plot.")
+
+    if server:
+        obj.session.store_document(obj.doc)
+        link = obj.session.object_link(obj.doc.context)
+        view(link)
+
+    if notebook:
+        from bokeh.embed import notebook_div
+        for plot in obj._plots:
+            publish_display_data({'text/html': notebook_div(plot)})
```

Comment:
wow, that's duplication of code duplication :stuck_out_tongue_winking_eye: 
maybe there is an easy and quick refactor to avoid that, don't you think?